python-threatexchange/threatexchange/fetcher/fetch_state.py

```diff
@@ -46,11 +66,78 @@ class SignalOpinion:
 
     owner: int
     category: SignalOpinionCategory
-    tags: t.List[str]
+    tags: t.Set[str]
+
+    @classmethod
+    def get_trivial(cls):
+        return cls(0, SignalOpinionCategory.WORTH_INVESTIGATING, [])
+
+
+class AggregateSignalOpinionCategory(IntEnum):
+    """
+    Represent multiple opinions as one.
+
+    Keep in Sync with SignalOpinionCategory
+    """
+
+    FALSE_POSITIVE = 0  # Signal generates false positives
+    WORTH_INVESTIGATING = 1  # Indirect indicator
+    TRUE_POSITIVE = 2  # Confirmed meets category
+    DISPUTED = 3  # Some positive, some negative
+
+    @classmethod
+    def from_opinion_categories(
+        cls, opinion_categories: t.Iterable[SignalOpinionCategory]
+    ) -> "AggregateSignalOpinionCategory":
+        aggregate_opinion = None
+        for category in opinion_categories:
+            aggregate_opinion = cls.aggregate(aggregate_opinion, category)
+        assert aggregate_opinion is not None
+        return aggregate_opinion
+
+    @classmethod
+    def aggregate(
+        cls,
+        old: t.Optional["AggregateSignalOpinionCategory"],
+        new: t.Union["AggregateSignalOpinionCategory", SignalOpinionCategory],
+    ) -> "AggregateSignalOpinionCategory":
+        """
+        Combine signal opinions into an aggregate opinion.
+
+        In general, take the highest confidence/severity of true positives,
+        unless you have both a true + false positive, in which case the result
+        is disputed.
+        """
+        # For some reason, mypy can't deal with new being retyped
```

Comment:
We've used t.cast for this in a lot of hmalib.

Reply:
TIL